tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -598,19 +596,23 @@ private void OnMvcBeforeAction(object arg)
                 {
                     if (!tracer.Settings.RouteTemplateResourceNamesEnabled)
                     {
-                        SetLegacyResourceNames(typedArg, parentSpan);
+                        // override the parent's resource name with the simplified MVC route template
+                        rootSpan.ResourceName = GetLegacyResourceName(typedArg);
                     }
                     else
                     {
-                        span = StartMvcCoreSpan(tracer, parentSpan, typedArg, httpContext, request);
+                        span = StartMvcCoreSpan(tracer, trackingFeature, typedArg, httpContext, request);
                     }
                 }
 
-                CurrentSecurity.CheckPathParamsFromAction(httpContext, span, typedArg.ActionDescriptor?.Parameters, typedArg.RouteData.Values);
+                if (span is not null)
```

Comment:
I think this had the potential to give a null ref from ASM as `span` isn't guaranteed to be non-null here